echopop/computation/biology.py

```diff
@@ -104,160 +128,751 @@ def sum_strata_weight( haul_data: pd.DataFrame ,
+        stratum_kriged_biomass.biomass_kriged
+        * stratum_kriged_biomass.proportion_weight_length_all 
+        * stratum_kriged_biomass.proportion_aged_weight_all
+    ).fillna( 0 )
+
+    # ---- Apportioned biomass for adult fish
+    stratum_kriged_biomass[ 'biomass_aged_adult' ] = (  
+        stratum_kriged_biomass.biomass_kriged
+        * stratum_kriged_biomass.proportion_weight_length_adult
+        * stratum_kriged_biomass.proportion_aged_weight_adult
+    ).fillna( 0 )
+
+    # ---- Sum across strata to produce 'grand totals' for all fish
+    apportioned_total_kriged_biomass = (
+        stratum_kriged_biomass
+        .groupby( [ 'species_id' , 'length_bin' , 'age_bin' ] )
+        .agg( { 'biomass_aged_all': 'sum' ,
+                'biomass_aged_adult': 'sum' } )
+        .reset_index( )
+    )
+
+    ### Return output (tuple)
+    return apportioned_sexed_kriged_biomass , apportioned_total_kriged_biomass
+
+def compute_unaged_number_proportions( length_data: pd.DataFrame ,
+                                       length_intervals: np.ndarray ):
+    """
+    Calculate length binned weight proportions among unaged fish  
+
+    Parameters
+    ----------
+    length_data: pd.DataFrame
+        Dataframe containing length data measured from unaged fish
+    length_intervals: np.ndarray
+        Array containing length bins/intervals
+    """ 
+
+    ### Calculate number proportion
+    # ---- Bin length measurements
+    length_data_binned = (
+        length_data
+        # ---- Bin length
+        .bin_variable( length_intervals , 'length' ) 
+    )
+
+    # ---- Sum the number of individuals within each bin 
+    proportions_unaged_length = (
+        length_data_binned
+        # ---- Group number summations across stratum/species/sex/length
+        .groupby( [ 'stratum_num' , 'species_id' , 'length_bin' ] )
+        # ---- Sum count
+        .agg( number_all = ( 'length_count' , 'sum' ) )
+        # ---- Fill empty/non-existent values with 0's
+        .fillna( 0 )
+        .reset_index( )
+    )
+
+    ### Calculate the number proportions
+    # --- Stratum total counts
+    proportions_unaged_length[ 'stratum_number_all' ] = (
+        proportions_unaged_length.groupby( [ 'stratum_num' ] )[ 'number_all' ].transform( sum )
+    )
+
+    # ---- Proportions of each sex-length bin pair relative to `stratum_number_all`
+    proportions_unaged_length[ 'proportion_number_all' ] = (
+        proportions_unaged_length.number_all / proportions_unaged_length.stratum_number_all
+    ).fillna( 0 )
+
+    ### Filter out unnecessary columns and return output
+    return proportions_unaged_length.filter( regex = '^(?!number_|stratum_number_).*' )
+
+def compute_unaged_weight_proportions( proportions_unaged_length: pd.DataFrame ,
+                                       length_weight_df: pd.DataFrame ):
+    """
+   Compute the unaged weight proportions across all fish and specific sexes for each
+   length bin  
```

Comment:
wording suggestions and also fixed indent -- let's use the code formatter!
```suggestion
    Compute the weight proportion of each length bin within each stratum for all unaged fish.
```